mlflow/models/dependencies_schemas.py

```diff
@@ -55,23 +54,31 @@ def set_retriever_schema(
     retriever_schema = globals().get(DependenciesSchemasType.RETRIEVERS.value, [])
 
     # Check if a retriever schema with the same name already exists
-    if any(schema["name"] == name for schema in retriever_schema):
-        # reset if there is an error to clear the global state for next run
-        _clear_retriever_schema()
-        raise MlflowException(
-            f"A retriever schema with the name '{name}' already exists.",
-            error_code=ALREADY_EXISTS,
+    existing_schema = next((schema for schema in retriever_schema if schema["name"] == name), None)
+
+    if existing_schema is not None:
+        warnings.warn(
+            f"A retriever schema with the name '{name}' already exists. "
+            "Overriding the existing schema.",
+            category=UserWarning,
+            stacklevel=2,
+        )
```

Comment:
```suggestion
       _logger.warning(
            f"A retriever schema with the name '{name}' already exists. "
            "Overriding the existing schema.",
        )
```

Can we use a logger to ensure that the warning is emitted every time `set_retriever_schema` is called?

Reply:
Sure. Do we know how we can test that the logger was fired? Or we don't need to test for that?

Reply:
We can mock `_logger.warning` and check it's fired.